In [1]:
import pm_evaluation

### Evaluation

We use gt for evaluations. We do blocking and pw on gt.

#### Blocking

In [2]:
import importlib.util
import sys

module_dir = "../pairwise_matching/ditto"
module_path = "../pairwise_matching/ditto/matcher.py"

# Add the directory to sys.path (to allow imports inside the module to work)
if module_dir not in sys.path:
    sys.path.append(module_dir)

# Load the module dynamically
module_name = "your_module_name"
spec = importlib.util.spec_from_file_location(module_name, module_path)
ditto_matcher = importlib.util.module_from_spec(spec)
spec.loader.exec_module(ditto_matcher)

module_dir = "../pairwise_matching"
module_path = "../pairwise_matching/rll_pairwise_matching.py"

# Add the directory to sys.path (to allow imports inside the module to work)
if module_dir not in sys.path:
    sys.path.append(module_dir)

# Load the module dynamically
module_name = "your_module_name"
spec = importlib.util.spec_from_file_location(module_name, module_path)
jw_matcher = importlib.util.module_from_spec(spec)
spec.loader.exec_module(jw_matcher)

c:\Program Files\Python311\Lib\site-packages\tqdm\auto.py:21: TqdmWarning: IProgress not found. Please update jupyter and ipywidgets. See https://ipywidgets.readthedocs.io/en/stable/user_install.html
  from .autonotebook import tqdm as notebook_tqdm


In [4]:
blocking_paths = ["lsh_bigram_blocking.json","lsh_bigram_gt_blocking.json", "lsh_words_gt_blocking.json", "lsh_bigram_aq_gt_blocking.json", "lsh_words_aq_gt_blocking.json" ]
pairwise_methods = ["jarowinkler", "ditto", "deepmatcher"]

for blocking_path in blocking_paths:
    if blocking_path == "lsh_bigram_blocking.json":
        pairs: list = pm_evaluation.get_pairs_for_pairwise_matching(blocking_path)
        print(len(pairs))
        
        for pw_method in pairwise_methods:
            #if pw_method == "jarowinkler":
                #for threshold in [0.7, 0.75]:
                    # do jarowinkler pw and save to file
                    #jw_matcher.pairwise_matching(blocking_path, threshold)
            if pw_method == "ditto":
                blocking_path = blocking_path.replace("json", "txt")
                ditto_matcher.pairwise_matching(pairs_to_evaluate=pairs,
                                                checkpoint_path='../pairwise_matching/ditto/checkpoints/',
                                                ditto_path='../pairwise_matching/ditto/',
                                                output_path='results/ditto/' + blocking_path)
            elif pw_method == "deepmatcher":
                # do dm pw aoooo
                pass

609295


c:\Code\VSCode\de-projects\project-5\src\evaluation\../pairwise_matching/ditto/matcher.py:305: FutureWarning: You are using `torch.load` with `weights_only=False` (the current default value), which uses the default pickle module implicitly. It is possible to construct malicious pickle data which will execute arbitrary code during unpickling (See https://github.com/pytorch/pytorch/blob/main/SECURITY.md#untrusted-models for more details). In a future release, the default value for `weights_only` will be flipped to `True`. This limits the functions that could be executed during unpickling. Arbitrary objects will no longer be allowed to be loaded via this mode unless they are explicitly allowlisted by the user via `torch.serialization.add_safe_globals`. We recommend you start setting `weights_only=True` for any use case where you don't have full control of the loaded file. Please open an issue on GitHub for any issues related to this experimental feature.
  saved_state = torch.load(checkpo

load_f1 = 0.9465648854961832
real_f1 = 0.9545454545454546


#### Eval of pairwise

In [4]:
import os
RED = "\033[31m"
RESET = "\033[0m"

for dirpath, _, filenames in os.walk("results"):
    if dirpath != "results":
        name = dirpath.split("\\")[1].upper()
        print(f"{RED}{name}{RESET}")
    for filename in filenames:
        filepath = os.path.join(dirpath, filename)
        pm_evaluation.evaluate("../ground_truth/gt.txt", filepath)
        

DEEPMATCHER

Pairwise matching EVALUATION for results\deepmatcher\lsh_bigram_gt_blocking.txt
- Precision: 0.99
- Recall: 0.83
- F1: 0.90


Pairwise matching EVALUATION for results\deepmatcher\lsh_words_gt_blocking.txt
- Precision: 0.99
- Recall: 0.84
- F1: 0.91

DITTO

Pairwise matching EVALUATION for results\ditto\lsh_bigram_aq_gt_blocking.txt
- Precision: 0.97
- Recall: 0.96
- F1: 0.96


Pairwise matching EVALUATION for results\ditto\lsh_bigram_gt_blocking.txt
- Precision: 0.97
- Recall: 0.98
- F1: 0.97


Pairwise matching EVALUATION for results\ditto\lsh_words_aq_gt_blocking.txt
- Precision: 0.97
- Recall: 0.95
- F1: 0.96


Pairwise matching EVALUATION for results\ditto\lsh_words_gt_blocking.txt
- Precision: 0.97
- Recall: 0.98
- F1: 0.97

JAROWINKLER

Pairwise matching EVALUATION for results\jarowinkler\lsh_bigram_aq_gt_t0.7.txt
- Precision: 0.96
- Recall: 0.92
- F1: 0.94


Pairwise matching EVALUATION for results\jarowinkler\lsh_bigram_aq_gt_t0.75.txt
- Precision: 0.96
- Recall: 0

#### Execution Times

##### Ditto
Time for ditto pairwise matching on 65.895 items - 609295 pairs after lsh_bigram_blocking: **71 minutes.**

Time for ditto pairwise matching on gt items with both blocking methods: **Average of 1.6 seconds**

##### Jarowinkler
Time for jarowinkler pairwise matching on 65.895 items after blocking: **Average of  25 seconds**